### ISSUE TITLE
NaN return value when finding the distance to the same point

### ISSUE THREAD

**Issue by bleege** [mapbox/mapbox-gl-native#3556]
With certain LatLng values (e.g. LatLng(40.71199035644531, -74.0081)) we are getting a NaN return value when finding the distance to the same point.

```java
LatLng ll1 = new LatLng(40.71199035644531, -74.0081);
LatLng ll2 = new LatLng(40.71199035644531, -74.0081);
double distance = ll1.distanceTo(ll2); // distance: NaN
```

It appears that in this particular case the sum of t1, t2 & t3 equals 1.0000000000000002 instead of 1 which then causes `Math.acos()` to return NaN since its > 1.

```java
    public double distanceTo(LatLng other) {

        final double a1 = MathConstants.DEG2RAD * this.latitude;
        final double a2 = MathConstants.DEG2RAD * this.longitude;
        final double b1 = MathConstants.DEG2RAD * other.getLatitude();
        final double b2 = MathConstants.DEG2RAD * other.getLongitude();

        final double cosa1 = Math.cos(a1);
        final double cosb1 = Math.cos(b1);

        final double t1 = cosa1 * Math.cos(a2) * cosb1 * Math.cos(b2);
        final double t2 = cosa1 * Math.sin(a2) * cosb1 * Math.sin(b2);
        final double t3 = Math.sin(a1) * Math.sin(b1);
        final double tt = Math.acos(t1 + t2 + t3);

        return GeoConstants.RADIUS_EARTH_METERS * tt;
    }
```

**Comment by tobrun** [comment]
I started with writing a testcase for this issue:
![screen shot 2016-01-18 at 12 30 29](https://cloud.githubusercontent.com/assets/2151639/12390340/4e35a714-bddf-11e5-9824-301ed05c5cda.png)

This resulted in following output:

![screen shot 2016-01-18 at 12 30 34](https://cloud.githubusercontent.com/assets/2151639/12390342/5771178c-bddf-11e5-92e1-a2d076185f9e.png)

**Comment by tobrun** [comment]
Updated code of `distanceTo`:

![screen shot 2016-01-18 at 12 41 59](https://cloud.githubusercontent.com/assets/2151639/12390514/eb80306a-bde0-11e5-9914-da88d8772507.png)

**Comment by tobrun** [comment]
This landed with 78bedff in release branch of `3.1.0` in #3537
